tests/integration/dataframe/test_dataframe.py

```diff
@@ -213,3 +211,61 @@ def test__modify_table__returns(self, client: DataFrameClient, create_table):
         assert table.name == id
         assert table.properties == {"bee": "buzz"}
         assert table.columns[0].properties == {}
+
+    def test__delete_table__deletes(self, client: DataFrameClient):
+        id = client.create_table(  # Don't use fixture to avoid deleting the table twice
+            models.CreateTableRequest(
+                columns=[
+                    models.Column(
+                        name="index",
+                        data_type=models.DataType.Int32,
+                        column_type=models.ColumnType.Index,
+                    )
+                ]
+            )
+        )
+
+        assert client.delete_table(id) is None
+
+        with pytest.raises(ApiException, match="404 Not Found"):
+            client.get_table_metadata(id)
+
+    def test__delete_tables__deletes(self, client: DataFrameClient):
+        ids = [
+            client.create_table(
+                models.CreateTableRequest(
+                    columns=[
+                        models.Column(
+                            name="index",
+                            data_type=models.DataType.Int32,
+                            column_type=models.ColumnType.Index,
+                        )
+                    ]
+                )
+            )
+            for _ in range(3)
+        ]
+
+        assert client.delete_tables(ids) is None
+
+        assert client.list_tables(id=ids).tables == []
+
+    def test__delete_tables__returns_partial_success(self, client: DataFrameClient):
+        id = client.create_table(
+            models.CreateTableRequest(
+                columns=[
+                    models.Column(
+                        name="index",
+                        data_type=models.DataType.Int32,
+                        column_type=models.ColumnType.Index,
+                    )
+                ]
+            )
+        )
+
+        response = client.delete_tables([id, "invalid_id"])
+
+        assert response is not None
+        assert response.deleted_table_ids == [id]
+        assert response.failed_table_ids == ["invalid_id"]
+        assert response.error is not None
```

Comment:
Could we also assert that there is a single inner error to make sure we're deserializing those correctly?